node/runtime/src/lib.rs

```diff
@@ -379,6 +379,20 @@ impl sudo::Trait for Runtime {
 	type Proposal = Call;
 }
 
+impl im_online::Trait for Runtime {
+	type AuthorityId = AuraId;
+	type Call = Call;
+	type Event = Event;
+	type SessionsPerEra = SessionsPerEra;
+	type UncheckedExtrinsic = UncheckedExtrinsic;
+
+	fn is_valid_authority_id(authority_id: &<Self as im_online::Trait>::AuthorityId) -> bool {
```

Comment:
this is irregular. instead, create a new `trait IsMember<AuthorityId>`, have `Aura` and `Babe` implement it and have the `im_online::Trait` contain an extra item `type IsValidAuthorityId: IsMember<Self::AuthorityId>;`. here there should just be a `type IsValidAuthorityId = Aura;`.